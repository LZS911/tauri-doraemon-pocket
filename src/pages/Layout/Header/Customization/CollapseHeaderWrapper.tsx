const CollapseHeaderWrapper: React.FC<{
  title: string;
  subTitle: string;
  icon: React.ReactNode;
}> = ({ title, icon, subTitle }) => (
  <div className="flex items-center">
    <div className="flex items-center justify-center rounded bg-secondary p-2 text-lg text-primary">
      {icon}
    </div>

    <div className="ml-5">
      <p>{title}</p>
      <p className="text-xs opacity-40">{subTitle}</p>
    </div>
  </div>
);

export default CollapseHeaderWrapper;
