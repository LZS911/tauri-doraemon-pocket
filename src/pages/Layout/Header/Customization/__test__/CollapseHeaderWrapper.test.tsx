import { render } from '@testing-library/react';
import CollapseHeaderWrapper from '../CollapseHeaderWrapper';

describe('test CollapseHeaderWrapper', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <CollapseHeaderWrapper
        title="title"
        subTitle="subTitle"
        icon={<>icon</>}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
