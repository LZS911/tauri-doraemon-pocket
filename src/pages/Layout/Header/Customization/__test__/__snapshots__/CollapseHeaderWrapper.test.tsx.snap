// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CollapseHeaderWrapper should match snapshot 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <div
      class="flex items-center justify-center rounded bg-secondary p-2 text-lg text-primary"
    >
      icon
    </div>
    <div
      class="ml-5"
    >
      <p>
        title
      </p>
      <p
        class="text-xs opacity-40"
      >
        subTitle
      </p>
    </div>
  </div>
</div>
`;
